clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/MigrateServerGroupStrategy.java

```diff
@@ -0,0 +1,271 @@
+
+    List<MigrateLoadBalancerResult> targetLoadBalancers = generateTargetLoadBalancers(
+      sourceGroup, source, target, subnetType, dryRun);
+
+    MigrateServerGroupResult migrateResult = new MigrateServerGroupResult();
+
+    List<MigrateSecurityGroupResult> targetSecurityGroups = generateTargetSecurityGroups(launchConfig, source,  target,
+      migrateResult, dryRun);
+
+    Map<String, List<String>> zones = new HashMap<>();
+    zones.put(target.getRegion(), target.getAvailabilityZones());
+
+    DeploymentResult result;
+    if (!dryRun) {
+      Capacity capacity = getCapacity();
+      BasicAmazonDeployDescription deployDescription = new BasicAmazonDeployDescription();
+      deployDescription.setSource(getSource(source));
+      deployDescription.setCredentials(target.getCredentials());
+      deployDescription.setAmiName(targetAmi != null ? targetAmi : launchConfig.getImageId());
+      deployDescription.setApplication(names.getApp());
+      deployDescription.setStack(names.getStack());
+      deployDescription.setFreeFormDetails(names.getDetail());
+      deployDescription.setInstanceMonitoring(launchConfig.getInstanceMonitoring().getEnabled());
+      deployDescription.setInstanceType(launchConfig.getInstanceType());
+      deployDescription.setIamRole(iamRole != null ? iamRole : launchConfig.getIamInstanceProfile());
+      deployDescription.setKeyPair(keyPair != null ? keyPair : launchConfig.getKeyName());
+      deployDescription.setAssociatePublicIpAddress(launchConfig.getAssociatePublicIpAddress());
+      deployDescription.setCooldown(sourceGroup.getDefaultCooldown());
+      deployDescription.setHealthCheckGracePeriod(sourceGroup.getHealthCheckGracePeriod());
+      deployDescription.setHealthCheckType(sourceGroup.getHealthCheckType());
+      deployDescription.setSuspendedProcesses(sourceGroup.getSuspendedProcesses().stream()
+        .map(SuspendedProcess::getProcessName).collect(Collectors.toSet()));
+      deployDescription.setTerminationPolicies(sourceGroup.getTerminationPolicies());
+      deployDescription.setKernelId(launchConfig.getKernelId());
+      deployDescription.setEbsOptimized(launchConfig.getEbsOptimized());
+      deployDescription.setBase64UserData(launchConfig.getUserData());
+      deployDescription.setLoadBalancers(targetLoadBalancers.stream()
+        .map(MigrateLoadBalancerResult::getTargetName).collect(Collectors.toList()));
+      deployDescription.setSecurityGroups(targetSecurityGroups.stream()
+        .map(sg -> sg.getTarget().getTargetName()).collect(Collectors.toList()));
+      deployDescription.setAvailabilityZones(zones);
+      deployDescription.setStartDisabled(true);
+      deployDescription.setCapacity(capacity);
+      deployDescription.setSubnetType(subnetType);
+
+      BasicAmazonDeployDescription description = generateDescription(source, deployDescription);
+
+      result = getBasicAmazonDeployHandler().handle(description, new ArrayList());
```

Comment:
may be worth injecting `basicAmazonDeployDescriptionValidator` here and validating the resulting description before trying to execute it